### ISSUE TITLE
Remote LSP projects

### ISSUE THREAD

**Issue by traxys** [miversen33/netman.nvim#146]
In some languages (like C, rust,...) you can't have a LSP on single file, you need to find the repo root and then execute some code according to that.

For example in C you need a compile_command.json, and the file contains a lot of absolute paths.

Is it possible to use the *remote* filesystem for this, or would you need a mirror of the remote repository on the local side just to have the correct file structure ?

**Comment by miversen33** [comment]
So the big issue with LSPs right now is that they are _external_ processes (IE, they live outside the neovim process). This means that netman doesn't have direct control of how the LSPs see the remote filesystem. 

Additionally, the underlying file isn't actually presented to external processes. The external processes will see the URI of the file instead of the underlying file name.

This is on my list of stuff to deal with, I just haven't fleshed out a great solution yet. I am absolutely open to ideas though! 

Please note though, getting LSPs to play nice is definitely a backlog item. I need to get Async completed (and probably finalize the neo-tree integration) before I feel comfortable looking deeper into this issue.

That said, I am more than happy to accept PRs :)

**Comment by miversen33** [comment]
I suppose I didn't really answer your question, just dumped some general info about why netman and LSPs don't play super great together.

To the actual question, in order to make the LSP scenario described above work, there needs to be some way for the external LSP process to "read" the remote file system. That doesn't necessarily mean the a duplicate of the remote needs to exist locally, just that the process can query a directory like structure that presents itself like the remote structure.

Fuse is a great option for this, and I'm currently exploring how that would look but it's a bit complicated and thus relegated to the backlog for now. Fuse also is a Linux only thing (I assume it works on Mac too so maybe it's a unix thing?) So I'd need to consider an option to work with Windows, as I do eventually want to make this thing work there too.

Anyway the long short, any kind of in-depth LSP stuff is a ways off from functioning with netman:/

**Comment by traxys** [comment]
Yeah I guessed that LSP was not ready at all, but I wanted to have some clarifications on what would and would not be supported right now

**Comment by miversen33** [comment]
Supported? Nothing at this time. 

What can you get working? Single file LSP support (stuff like what python, java, JavaScript, etc provide) _probably_ works but I wouldn't rely on it.

It's on my list :)

**Comment by traxys** [comment]
Thank you very much for the clarifications! It was not imposing any kinds of expectations, just intersted in the plugin & what I can do with it today!

**Comment by miversen33** [comment]
I didn't read it as such :) If you have further questions, please let me know! I am still working through updating the documentation and I understand it can be a bit daunting looking at it.

I am keeping this issue open as the technical request (Remote LSP Projects) is one that does need addressing _eventually_. I've added it to the backlog and maybe someday before the heat death of the universe, I will be able to get around to it :)
